From ff598ea3c17e047a2f7f95f213f32c1144c1155a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Fri, 11 May 2012 09:17:48 +0300
Subject: [PATCH 2/5] Fix install targets

QtWebKit has yet another abuse of DESTDIR and "copy to system paths
during build" evilness. Remove them and rely on install targets alone,
so the module can build in modular fashion in an autobuilder.
---
 Source/WebKit/qt/declarative/experimental/experimental.pri |   12 ------------
 Source/WebKit/qt/declarative/public.pri                    |   12 ------------
 2 files changed, 24 deletions(-)

diff --git a/Source/WebKit/qt/declarative/experimental/experimental.pri b/Source/WebKit/qt/declarative/experimental/experimental.pri
index 583726f..c26fe33 100644
--- a/Source/WebKit/qt/declarative/experimental/experimental.pri
+++ b/Source/WebKit/qt/declarative/experimental/experimental.pri
@@ -11,25 +11,13 @@ TARGET.module_name = QtWebKit/experimental
 
 CONFIG += qt plugin
 
-QMLDIRFILE = $${_PRO_FILE_PWD_}/qmldir
-copy2build.input = QMLDIRFILE
-copy2build.output = $${ROOT_BUILD_DIR}/imports/$${TARGET.module_name}/qmldir
-!contains(TEMPLATE_PREFIX, vc):copy2build.variable_out = PRE_TARGETDEPS
-copy2build.commands = $$QMAKE_COPY ${QMAKE_FILE_IN} ${QMAKE_FILE_OUT}
-copy2build.name = COPY ${QMAKE_FILE_IN}
-copy2build.CONFIG += no_link
-QMAKE_EXTRA_COMPILERS += copy2build
-
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
 
 wince*:LIBS += $$QMAKE_LIBS_GUI
 
 QT += declarative widgets network quick webkit webkit-private
 
-DESTDIR = $${ROOT_BUILD_DIR}/imports/$${TARGET.module_name}
-
 CONFIG += rpath
-RPATHDIR_RELATIVE_TO_DESTDIR = ../../lib
 
 SOURCES += plugin.cpp
 
diff --git a/Source/WebKit/qt/declarative/public.pri b/Source/WebKit/qt/declarative/public.pri
index e389e38..d938fa2 100644
--- a/Source/WebKit/qt/declarative/public.pri
+++ b/Source/WebKit/qt/declarative/public.pri
@@ -13,15 +13,6 @@ CONFIG += qt plugin
 
 load(features)
 
-QMLDIRFILE = $${_PRO_FILE_PWD_}/qmldir
-copy2build.input = QMLDIRFILE
-copy2build.output = $${ROOT_BUILD_DIR}/imports/$${TARGET.module_name}/qmldir
-!contains(TEMPLATE_PREFIX, vc):copy2build.variable_out = PRE_TARGETDEPS
-copy2build.commands = $$QMAKE_COPY ${QMAKE_FILE_IN} ${QMAKE_FILE_OUT}
-copy2build.name = COPY ${QMAKE_FILE_IN}
-copy2build.CONFIG += no_link
-QMAKE_EXTRA_COMPILERS += copy2build
-
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
 
 wince*:LIBS += $$QMAKE_LIBS_GUI
@@ -34,10 +25,7 @@ contains(DEFINES, HAVE_QQUICK1=1) {
     HEADERS += qdeclarativewebview_p.h
 }
 
-DESTDIR = $${ROOT_BUILD_DIR}/imports/$${TARGET.module_name}
-
 CONFIG += rpath
-RPATHDIR_RELATIVE_TO_DESTDIR = ../../lib
 
 # FIXME: Why are these needed, and why can't we use WEBKIT += ... ?
 INCLUDEPATH += \
-- 
1.7.10

